dawn/renderer/components/Editor.js

```diff
@@ -134,48 +151,90 @@ class Editor extends React.Component {
     setTimeout(() => this.refs.CodeEditor.editor.resize(), 0.1);
   }
 
-  sendCode(command) {
+  upload() {
+    const filepath = this.props.filepath;
+    if (filepath == null) {
+      dialog.showMessageBox({
+        type: 'warning',
+        buttons: ['Close'],
+        title: 'No Files',
+        message: 'No file? No upload',
+      });
+      console.log('No file? No upload');
+      return;
+    }
     const correctedText = this.correctText(this.props.editorCode);
     if (correctedText !== this.props.editorCode) {
       this.props.onAlertAdd(
         'Invalid characters detected',
         'Your code has non-ASCII characters, which won\'t work on the robot. ' +
-        'Please remove them and try again.'
+        'Please remove them and try again.',
       );
-      return false;
+      return;
     }
-    console.log('Deprecated: sendCode');
-    console.log(`Debug Info: ${command} EOD`);
-    return true;
-  }
-
-  upload() {
-    this.sendCode('upload');
+    const conn = new Client();
+    conn.on('error', (err) => {
+      dialog.showMessageBox({
+        type: 'warning',
+        buttons: ['Close'],
+        title: 'Connection Issue',
+        message: 'Could Not Connect to Robot',
+      });
+      throw err;
+    });
+    conn.on('ready', () => {
+      conn.sftp((err, sftp) => {
+        if (err) {
+          dialog.showMessageBox({
+            type: 'warning',
+            buttons: ['Close'],
+            title: 'Connection Issue',
+            message: 'Could Not Connect to Robot',
+          });
+          throw err;
+        }
+        console.log('SSH Connection');
+        sftp.fastPut(filepath, './studentcode/studentcode.py', (err2) => {
+          if (err2) {
+            dialog.showMessageBox({
+              type: 'warning',
+              buttons: ['Close'],
+              title: 'Upload Issue',
+              message: 'Code Upload Failed.',
+            });
+            throw err2;
+          }
+        });
+      });
+    }).connect({
+      debug: (inpt) => { console.log(inpt); },
+      host: this.props.ipAddress,
+      port: 22,
+      username: 'ubuntu',
```

Comment:
So you have the port, username, and password hardcoded both here and in another file later on. Any chance you could move to a single source of truth for host/port/username/password?

Besides avoiding the two places in code falling out of sync, I have some simulator-related ideas that would involve replacing these hardcoded constants with configurable values.